### ISSUE TITLE
Workflow introspection missing

### ISSUE THREAD

**Issue by zopyx** [plone/plone.restapi#1003]
Not a direct feature request but a common observation in working with plone.restapi:

Using the `@workflow` endpoint requires knowledge about the underlying workflow and its transitions. 
In particular with custom workflows, it is very hard to write custom code for bringing a particular object into a particular workflow state (e.g. as part of a migration). 

Perhaps two options:
- allow introspection on all workflows
- perhaps useful: some way to set the workflow state and the related object security directly (as asked in my question https://community.plone.org/t/setting-review-state-quick-programmatically/12991)

**Comment by tisto** [comment]
@zopyx good point! We would need to draft the API design and then work on the implementation.
